test/e2e/specs/editor/blocks/navigation.spec.js

```diff
@@ -146,3 +146,39 @@ class NavigationBlockUtils {
 		);
 	}
 }
+
+test.describe( 'Navigation block', () => {
+	test.describe(
+		'As a user I want to see a warning if the menu referenced by a navigation block is not available',
+		() => {
+			test.beforeEach( async ( { admin } ) => {
+				await admin.createNewPost();
+			} );
+
+			test( 'warning message shows when given an unknown ref', async ( {
+				editor,
+			} ) => {
+				await editor.insertBlock( {
+					name: 'core/navigation',
+					attributes: {
+						ref: 1,
+					},
+				} );
+
+				// Check the markup of the block is correct.
+				await editor.publishPost();
+
+				const content = await editor.getEditedPostContent();
+				expect( content ).toBe( `<!-- wp:navigation {"ref":1} /-->` );
```

Comment:
Nit: `editor.getBlocks` is preferred as shown in a recent [doc](https://github.com/WordPress/gutenberg/blob/trunk/docs/contributors/code/e2e/overusing-snapshots.md#the-solution) for slightly improved readability.

```suggestion
				await expect.poll( editor.getBlocks ).toMatchObject( [ {
					name: 'core/navigation',
					attributes: { ref: 1 },
				} ] );
```